aws/resource_aws_ec2_transit_gateway_peering_attachment_test.go

```diff
@@ -0,0 +1,413 @@
+
+		if err != nil {
+			return err
+		}
+
+		if attachment == nil {
+			return fmt.Errorf("EC2 Transit Gateway Peering Attachment not found")
+		}
+
+		if aws.StringValue(attachment.State) != ec2.TransitGatewayAttachmentStateAvailable && aws.StringValue(attachment.State) != ec2.TransitGatewayAttachmentStatePendingAcceptance {
+			return fmt.Errorf("EC2 Transit Gateway Peering Attachment (%s) exists in non-available/pending acceptance (%s) state", rs.Primary.ID, aws.StringValue(attachment.State))
+		}
+
+		*transitGatewayPeeringAttachment = *attachment
+
+		return nil
+	}
+}
+
+func testAccCheckAWSEc2TransitGatewayPeeringAttachmentDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).ec2conn
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_ec2_transit_gateway_peering_attachment" {
+			continue
+		}
+
+		peeringAttachment, err := ec2DescribeTransitGatewayPeeringAttachment(conn, rs.Primary.ID)
+
+		if isAWSErr(err, "InvalidTransitGatewayAttachmentID.NotFound", "") {
+			continue
+		}
+
+		if err != nil {
+			return err
+		}
+
+		if peeringAttachment == nil {
+			continue
+		}
+
+		if aws.StringValue(peeringAttachment.State) != ec2.TransitGatewayAttachmentStateDeleted {
+			return fmt.Errorf("EC2 Transit Gateway Peering Attachment (%s) still exists in non-deleted (%s) state", rs.Primary.ID, aws.StringValue(peeringAttachment.State))
+		}
+	}
+
+	return nil
+}
+
+func testAccCheckAWSEc2TransitGatewayPeeringAttachmentDisappears(transitGatewayPeeringAttachment *ec2.TransitGatewayPeeringAttachment) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		conn := testAccProvider.Meta().(*AWSClient).ec2conn
+
+		input := &ec2.DeleteTransitGatewayPeeringAttachmentInput{
+			TransitGatewayAttachmentId: transitGatewayPeeringAttachment.TransitGatewayAttachmentId,
+		}
+
+		if _, err := conn.DeleteTransitGatewayPeeringAttachment(input); err != nil {
+			return err
+		}
+
+		return waitForEc2TransitGatewayPeeringAttachmentDeletion(conn, aws.StringValue(transitGatewayPeeringAttachment.TransitGatewayAttachmentId))
+	}
+}
+
+func testAccAWSEc2TransitGatewayPeeringAttachmentConfig_base(rName string) string {
+	return fmt.Sprintf(`
+resource "aws_ec2_transit_gateway" "test" {
+  tags = {
+    Name = %[1]q
+  }
+}
+
+resource "aws_ec2_transit_gateway" "peer" {
+  provider = "aws.alternate"
+
+  tags = {
+    Name = %[1]q
+  }
+}
+`, rName)
+}
+
+func testAccAWSEc2TransitGatewayPeeringAttachmentConfig_sameAccount_base(rName string) string {
+	return testAccAlternateRegionProviderConfig() + testAccAWSEc2TransitGatewayPeeringAttachmentConfig_base(rName)
+}
+
+func testAccAWSEc2TransitGatewayPeeringAttachmentConfig_differentAccount_base(rName string) string {
+	return testAccAlternateAccountAlternateRegionProviderConfig() +
+		testAccAWSEc2TransitGatewayPeeringAttachmentConfig_base(rName) +
+		fmt.Sprintf(`
+data "aws_caller_identity" "peer" {
+  provider = "aws.alternate"
+}
+`)
+}
```

Comment:
With below simplification:

```suggestion
func testAccAWSEc2TransitGatewayPeeringAttachmentConfig_differentAccount_base(rName string) string {
	return testAccAlternateAccountAlternateRegionProviderConfig() + testAccAWSEc2TransitGatewayPeeringAttachmentConfig_base(rName)
}
```

Please note we also now have the `composeConfig()` function which accepts any number of strings to concatenate.